### PR TITLE
Added Google support

### PR DESCRIPTION
**Pull request by rob-mccann**
Hey,

I added Google support.

Notable changes:
Google only supports POST requests to get the token, so I added a parameter that defaults to GET but is overridden in Provider_Google as POST. The request is then posted in the access() function.

Added 'grant_type' => 'authorization_code' to the params when getting a token. I think this is in the OAuth spec. Removed 'response_type'.

Let me know if I've made any whoopsies.

(if this gets merged, I'll pull request the necessary change to ninjauth)

Rob :)

### CODE REVIEW

**Comment by philsturgeon** [comment]
Facebook requires response_type: http://d.pr/ujUf

Google also requires it: http://d.pr/h9G8

Read: http://d.pr/h9G8

**Comment by rob-mccann** [comment]
They require it for authorize(), but (correct me if I'm wrong) they don't require it for access().

File: classes/provider/google.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		
+		//normalise email
+		$primary_email = null;
+		
+		if(isset($user->emails))
+		{
+			//sometimes the G+ api gives us the emails as an array
+			foreach($user->emails as $email)
+			{
+				if($email->primary)
```
Comment (philsturgeon) [inline classes/provider/google.php:42]: { and } on new lines please :)

**Comment by philsturgeon** [comment]
If it works then great, make sure you double check Facebook is still 
kicking and we're good to go. I'll sort it out if I find it breaking on 
other providers.

Please make the change for ninjauth too :)

> ---
> 
>   unforeseen 
> mailto:reply@reply.github.com
> 11 October 2011 12:59
> 
> They require it for authorize(), but (correct me if I'm wrong) they 
> don't require it for access().
> 
> ---
> 
>   unforeseen 
> mailto:reply@reply.github.com
> 11 October 2011 12:50
> 
> Hey,
> 
> I added Google support.
> 
> Notable changes:
> Google only supports POST requests to get the token, so I added a 
> parameter that defaults to GET but is overridden in Provider_Google as 
> POST. The request is then posted in the access() function.
> 
> Added 'grant_type' => 'authorization_code' to the params when getting 
> a token. I think this is in the OAuth spec. Removed 'response_type'.
> 
> Let me know if I've made any whoopsies.
> 
> (if this gets merged, I'll pull request the necessary change to ninjauth)
> 
> Rob :)
> 
> You can merge this Pull Request by running:
> 
> git pull https://github.com/unforeseen/fuel-oauth2 develop/google
> 
> Or you can view, comment on it, or merge it online at:
> 
> https://github.com/fuel-packages/fuel-oauth2/pull/3
> 
> -- Commit Summary --
> - Added Google support
> 
> -- File Changes --
> 
> M bootstrap.php (1)
> M classes/provider.php (48)
> A classes/provider/google.php (83)
> 
> -- Patch Links --
> 
> https://github.com/fuel-packages/fuel-oauth2/pull/3.patch
> https://github.com/fuel-packages/fuel-oauth2/pull/3.diff

**Comment by rob-mccann** [comment]
Changed the naughty if-statements.

Facebook is working my end :)

File: classes/provider.php [hunk 2]
```diff
@@ -43,6 +43,11 @@ abstract class Provider {
 	 * @var  array  additional request parameters to be used for remote requests
 	 */
 	protected $params = array();
+	
+	/**
+	 * @var  string  the method to use when requesting tokens
+	 */
+	protected $method = 'GET';
```
Comment (philsturgeon) [inline classes/provider.php:50]: Last thing, is this property potentially a little ambiguous? Will authentication method need to be changed at all?

Comment (rob-mccann) [inline classes/provider.php:50]: I think I've got my publics/protecteds mixed up a bit here.

It gets changed in Provider_Google class (line 11)

Comment (philsturgeon) [inline classes/provider.php:50]: I meant is $method descriptive enough?

Comment (rob-mccann) [inline classes/provider.php:50]: how about token_request_method ?

Comment (philsturgeon) [inline classes/provider.php:50]: Works for me.

**Pull request merged**